feat(profiling) add transaction and % change text above toolbar

JonasBa opened the pull request:
![CleanShot 2023-12-05 at 12 14 25@2x](https://github.com/getsentry/sentry/assets/9317857/65d9a1a8-5cc7-43ec-bd27-1c0f2ef9c2ea)

JonasBa committed:
feat(profiling): add differential flamegraph txn

Zylphrex commented on static/app/components/events/eventStatisticalDetector/eventDifferentialFlamegraph.tsx:
```diff
@@ -54,20 +58,25 @@ export function EventDifferentialFlamegraph(props: EventDifferentialFlamegraphPr
+    EventsResultsDataRow<string> | undefined
+  >(undefined);
+
+  if (transaction === undefined) {
+    const firstTransaction = transactions?.data?.data?.[0];
+    if (firstTransaction) {
+      setTransaction(firstTransaction);
+    }
+  }
```
This part isn't very nice. We could potentially break it into 2 components where the outer component guarantees the transaction name is set before rendering the inner component to avoid this.

JonasBa replied:
Agree. I'll clean this up in a separate PR after I merge this last chain of changes

JonasBa committed:
feat(profiling): add txn pagination to toolbar (#61150)

https://github.com/getsentry/sentry/assets/9317857/59eaf27c-b966-413c-8e57-7e00ea7d1823
